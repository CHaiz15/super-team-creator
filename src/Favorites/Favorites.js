import React from 'react'
import './Favorites.scss'
import FavoriteCard from '../FavoriteCard/FavoriteCard'
import { connect } from 'react-redux'

export const Favorites = ({favorites}) => {
  const favoriteCards = favorites.map(favoriteTeam => <FavoriteCard team={favoriteTeam} />)
  return (
    <section className='fav-cards-container'>
      <h1 className='fav-title'>Favorites</h1>
      {favoriteCards}
    </section>
  )
}

export const mapStateToProps = state => ({
  favorites: state.favorites,
})

export default connect(mapStateToProps)(Favorites);

