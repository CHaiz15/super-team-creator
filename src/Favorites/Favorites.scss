@import '../variables';

.fav-title {
  @include mainTitles();
  margin: 0 0 10% 0;
  text-align: center;
}

.fav-cards-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  overflow: scroll;
}

.no-favorites {
  color: white;
}

.hidden {
  display: none;
}