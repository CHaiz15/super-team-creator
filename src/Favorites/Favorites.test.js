import React from 'react';
import { Favorites} from './Favorites.js';
import { shallow } from 'enzyme'

describe('Favorites', () => {
  let wrapper;
  let mockFavorites;
  beforeEach(() => {
      mockFavorites = [{
        name:'Best Team Ever',
        memberOne: {
          name:'Hulk', 
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        memberTwo: {
          name:'Thanos',
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        memberThree: {
          name:'Batman',
          powerstats: {
            Intelligence: 81,
            strength: 40,
            Speed: 29,
            Durability: 55,
            Power: 63,
            Combat: 90,
          },
          image: {url: "https://www.superherodb.com/pictures2/portraits/10/100/83.jpg"},
        },
        id: 123,
      }]
    wrapper = shallow(<Favorites favorites={mockFavorites}/>);
  })  
  it('should be an instance of the Favorites component', () => {
      expect(wrapper).toMatchSnapshot();
  });
})
