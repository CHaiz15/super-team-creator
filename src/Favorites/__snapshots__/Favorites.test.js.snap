// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should be an instance of the Favorites component 1`] = `ShallowWrapper {}`;
